### PR TITLE
Making more strict the user restriction to access forum and ensuring discussion_messages_controller as well

### PR DESCRIPTION
**Pull request by luchotc**
# :dart: Goal

This one changes the current behavior of discusser pseudo-permission. Nowadays you can always access the forum provided that the organization enables the `forum_enabled?` flag. If you don't reach that permission, you can access the forum but can't create discussions.
This is a bit inconsistent because as a student you are not prompted with the button to access the forum in the help box in assignments. The only link to the forum is the button that encourages a student to help others. I think with the current flow, it's better to deny the access altogether.

When a better flow is created and similar discussions are proposed, a student who can't create discussions may be offered with similar discussions in which it can't take part but may benefit. 

Also, we are allowing everyone to post messages, which isn't right :confused: 
 
 
# :arrow_backward: Retro-compatibility

This PR is fully backwards compatible code-wise.

# :eyes:  See also

https://github.com/mumuki/mumuki-domain/pull/111

### CODE REVIEW

File: app/controllers/discussions_controller.rb [hunk 1]
```diff
@@ -71,14 +66,6 @@ def discussion_filter_params
     @filter_params ||= params.permit(Discussion.permitted_query_params)
   end
 
-  def validate_forum_enabled!
```
Comment (flbulgarelli) [inline app/controllers/discussions_controller.rb:74]: :+1:

File: app/helpers/discussions_helper.rb [hunk 2]
```diff
@@ -23,8 +23,9 @@ def user_discussions_icon(text)
     fixed_fa_icon 'comment', text: text
   end
 
-  def discussions_link(item, path, html_options = nil, organization = Organization.current)
-    link_to item, path, html_options if organization.forum_enabled?
+  def discussions_link(item, path, html_options = nil)
+    return unless current_user&.can_discuss_here?
```
Comment (flbulgarelli) [inline app/helpers/discussions_helper.rb:27]: I don't have a strong position here, but is there any reason to prefer this validation style to the other?

File: spec/features/menu_bar_spec.rb [hunk 3]
```diff
@@ -52,15 +55,91 @@
...
+        private_organization.update! forum_enabled: true
+        private_organization.update! forum_only_for_trusted: true
+
+        visit '/'
+        expect(page).to have_text('Profile')
+        expect(page).not_to have_text('Classroom')
+        expect(page).not_to have_text('Bibliotheca')
+        expect(page).to have_text('Solve other\'s doubts')
+        expect(page).to have_text('My doubts')
+      end
```
Comment (flbulgarelli) [inline spec/features/menu_bar_spec.rb:142]: :heart:

File: app/helpers/discussions_helper.rb (hunk 2 above)
Comment (luchotc) [reply app/helpers/discussions_helper.rb:27]: I don't have a strong position either, but generally we are using this style when it comes to permission related actions, so it seemed more natural to do it this way

**Pull request merged**